### ISSUE TITLE
onChange return string instead of an obj

### ISSUE THREAD

**Issue by Angelk90** [downshift-js/downshift#770]
- `downshift` version: 3.3.1

**What you did**:

I'm creating the following situation using material-ui and Downshift.

![image](https://user-images.githubusercontent.com/20476002/65140033-f61c7180-da0d-11e9-940e-5a050a4c4f19.png)

**Problem description**:

Using the props onChange, su:

```
<Downshift
id="downshift-options"
 onChange={e => console.log(e)}
...
```

I return a string, that is the label (the one you see in the Chip), what I would like to return would be an obj, that is all the information of the selected user.

**Reproduction repository**:

Link: [codesandbox](https://codesandbox.io/s/material-demo-glf0u)

**Comment by silviuavram** [comment]
The behavior is expected because you are passing a string as an item

```
itemProps: getItemProps({
  item:
    suggestion.name +
    " " +
    suggestion.surname +
    " (" +
    suggestion.email +
    ")"
}),
```

Pass the whole item object instead like `item: suggestion` and you will get the object instead. And don't forget about the `itemToString` prop which is used to compute the string equivalent of the selectedItem and add it to your input automatically via `getInputProps`.

Good luck!

**Comment by Angelk90** [comment]
@silviuavram : 

Thank you, I hadn't thought of it, it seems to work, but I didn't understand the thing about itemToString, how it should be used.

Why the following message comes back to me:

`downshift: An object was passed to the default implementation of `itemToString`. You should probably provide your own `itemToString` implementation. Please refer to the `itemToString` API documentation. The object that was passed: `

Could you let me see if I don't disturb you how to do it?

Link: [codesandbox](https://codesandbox.io/s/material-demo-v0y3x)

P.s.

I would like to ask you for a suggestion on something, I should also do so that I already have a selected initial element, in some cases.
Element is passed as obj so it will be such a thing:

`{name: "Russell", surname: "Tran", address: "347 Lawn Court, Homeland, Wyoming, 8885", email: "russell.tran@apexia.ca"}`

I would like to know your opinion if you think the approach I am using is correct or if there is a better one.

**Comment by silviuavram** [comment]
On the first issue, please check the documentation related to `itemToString`. It needs to be a function like ``(suggestion) => `${suggestion.name} ${suggestion.surname} ${suggestion.email}` ``.
https://github.com/downshift-js/downshift#itemtostring

Once you provide that function you don't need to set the content manually to the `input` every time you select an item from the list. `getInputProps` will do that automatically for you.

On the second question, use `initialSelectedItem` prop and pass it the object you want to init your dropdown with. You can check that in the documentation as well. https://github.com/downshift-js/downshift#initialselecteditem

**Comment by Angelk90** [comment]
@silviuavram :

I'm not sure I understood how to use it and where to use it itemToString, at the moment I have something like this:
```
             <IntegrationDownshift
               options={users}
               onSearch={this.onSearch}
               onChange={(userSelect) => this.setState({userSelect})}
               selectedItem={userSelect}
               renderMenuItemChildren={option =>
                 `${option.name} ${option.surname}`
               }
               labelKey={option =>
                 `${option.name} ${option.surname} (${option.id})`
               }
               placeholder={"Cerca"}
               label={"User"}
/>
```

If I understand correctly on the other side I must have something like this:

```
 <Downshift
      id="downshift-options"
      onChange={handleChange}
      selectedItem={selectedItem}
      itemToString={item => (item ? item.value : '')}
  >

```

The problem that I have of items that vary.

1) 
```
{
name:"",
surname: "",
...
}
```

2) 
```
{
codeGroup: "",
infoGroup: "",
...
}
```


So there is something escaping me.

**Comment by silviuavram** [comment]
So, even if you have items that vary, you will handle that, you will conditionally render a string out of anything.

And you already do that in `getOptionLabel`.

What I am proposing is this:
- when you select an item, you put a big button Chip with the content `getOptionLabel(selectedItem)` on top of your input. You can try something else.
- instead, remove that `Chip` and pass `itemToString` to Downshift to be `suggestion => getOptionLabel(suggestion)`. You know better how to use `getOptionLabel` but it gives you a string equivalent of your object suggestion.
- by doing that, when you select an item, the `getOptionLabel(selectedItem)` becomes the new value of the input automatically, through `getInputProps`.

But now you still got the input, you can remove characters from it, you can Escape in it to remove the text and start the process all over again. And instead of that big Chip you can just render a Clear 'X' button next to the input that will have the `clearSelection` functionality.

**Comment by silviuavram** [comment]
Oh yeah and `itemToString` is a `Downshift` prop not a children function prop.

```jsx
      <Downshift
        id="downshift-options"
        onChange={handleChange}
        selectedItem={selectedItem}
        itemToString={item => 'bla'}
      >
```

**Comment by Angelk90** [comment]
@silviuavram :

I'm getting confused, are you advising to change the ui into something like that?

![Screenshot_20190919_100929](https://user-images.githubusercontent.com/20476002/65225620-bfa22d80-dac5-11e9-90e0-7b7474232806.png)

**Comment by silviuavram** [comment]
Keeping the input is what a combobox is doing.

But if keeping that Chip button is actually the UI you want, then at least pass the `itemToString` so you get rid of the warnings.

**Comment by silviuavram** [comment]
https://codesandbox.io/s/fls5o

Check `itemToString` usage in our `useSelect` hook.

We have objects as items. But we need the string equivalent for each in case we select an item.

If you open your screen reader (VoiceOver, Jaws, NVDA etc) and select an item, you will hear `Vacation has been selected` if you select vacation. That works because the itemToString result from the vacation object results in Vacation.

We don't have this message with the `Downshift` you are currently using, but we still use `itemToString` to put the string equivalent of the object you selected in the `input`.

Since you don't want to keep the input after selection that's fine, but we don't assume you wanted to do that and that is why you get warnings to pass us an `itemToString` in the case you are passing us Object items.

Does it make sense now? :)

**Comment by Angelk90** [comment]
@silviuavram : 
If you see here: 

Link: [codesandbox](https://qq38r.csb.app/)
I did so:
```
selectedItem={selectedItem}
itemToString={item => (item ? item.value : search)}
```

What I don't understand but in itemToString item is always `null`, what values should it have?
If not I could simply do so:

`        itemToString={item => search}
`

This only works if in IntegrationDownshift I have declared selectedItem, otherwise the text does not appear.

Does not seem to work properly, where am I doing wrong?
